### ISSUE TITLE
Improve `TestPrepareProposalConsistency`

### ISSUE THREAD

**Issue by evan-forbes** [celestiaorg/celestia-app#2519]
The `TestPrepareProposalConsistency` test uses randomly sized transactions to test if PrepareProposal will ever create a proposal block that ProcessProposal rejects. While other tests are responsible for ensuring that a transaction that gets through CheckTx, gets through PrepareProposal, and then ProcessProposal, this test can still be improved upon by being more exact with how many transactions are expected to be included in the block.

https://github.com/celestiaorg/celestia-app/blob/b59e876cd15ea9f17608d1c2ca865a86151b7e5f/app/test/fuzz_abci_test.go#L144

**Comment by ThanhNhann** [comment]
Hey @evan-forbes, can I handle this issue?

**Comment by exp7l** [comment]
Greetings! @evan-forbes @rootulp 
Looks like this is still open. Can I work on it?

**Comment by exp7l** [comment]
Looking

**Comment by rootulp** [comment]
Closing this as won't do because it's not immediately clear what this issue is asking for. See https://github.com/celestiaorg/celestia-app/pull/3087#issuecomment-1932966792

cc: @evan-forbes please re-open with acceptance criteria if we still want this.

**Comment by evan-forbes** [comment]
>  this test can still be improved upon by being more exact with how many transactions are expected to be included in the block.

I think this is the core piece, we aren't actually checking how many transactions are being included, only that one blobTx did. This means that the rest of the blob txs could be filtered out. Ideally we would make this more restrictive. This might involve writing a different or doing some analysis for what conservative % of txs should be expected to get through.

https://github.com/celestiaorg/celestia-app/blob/949a2d579619512e9e56eea10538fe847f12306d/app/test/fuzz_abci_test.go#L149-L155

**Comment by rawbytedev** [comment]
Hi
Looks like this is still open. I would like to work on it

**Comment by rawbytedev** [comment]
@evan-forbes 
> This means that the rest of the blob txs could be filtered out. Ideally we would make this more restrictive. This might involve writing a different or doing some analysis for what conservative % of txs should be expected to get through.

Would that mean that not all random blob are valid? But we are expecting at least 1 blob to be included 
So the improvement should be to set a min rate of blob that should be included in block for test to be valid

**Comment by rawbytedev** [comment]
# This the result of test:
https://github.com/rawbytedev/celestia-app/actions/runs/18431780761/job/52520034678
the first test gives over 10 %
but the second drops to 7%:
https://github.com/rawbytedev/celestia-app/actions/runs/18431780761/job/52520415905

We can expect to get 5% or higher of blob included in each of tests case

**Comment by rootulp** [comment]
It seems fragile to assert that > 5% of blobs get included in a block. I don't see the point of such a test and it seems likely to flake.

**Comment by rawbytedev** [comment]
Yes It still an assumptions and I'm not sure we can get a precise % of min included blob in each block. Maybe logging the results and noting how it evolve with time but other than that i  dunno what we could do

**Comment by rootulp** [comment]
One idea is to modify the fuzzing to create a fixed number of PFBs and send Txs and then in the test verify that the fixed number of PFBs and send Txs were included in the block.

**Comment by rawbytedev** [comment]
That would work but I think adding it as a complementary test will be better since with random blobs we are essentially looking for weird or edges cases (fuzzing)
Something like:
`TestPrepareProposalInclusion` can be used as additional test after `TestPrepareProposalConsistency`

**Comment by rawbytedev** [comment]
@rootulp

| Test Case                                         | Transaction Count | Valid Blobs | Generated Blobs (n_blob) | Inclusion Rate |
|---------------------------------------------------|------------------|-------------|--------------------------|----------------|
| many small single share single blob transactions  | 300              | 200         | 500                      | 40%            |
| one hundred normal sized single blob transactions | 200              | 100         | 100                      | 100%           |
| many single share multi-blob transactions         | 300              | 200         | 1000                     | 20%            |
| one hundred normal sized multi-blob transactions  | 200              | 100         | 100                      | 100%           |
| many small single share single blob transactions#01 | 300            | 200         | 500                      | 40%            |
| one hundred normal sized single blob transactions#01 | 200            | 100         | 100                      | 100%           |
| many single share multi-blob transactions#01      | 300              | 200         | 1000                     | 20%            |
| one hundred normal sized multi-blob transactions#01 | 200            | 100         | 100                      | 100%           |
| many small single share single blob transactions#02 | 300            | 200         | 500                      | 40%            |
| one hundred normal sized single blob transactions#02 | 200            | 100         | 100                      | 100%           |
| many single share multi-blob transactions#02      | 300              | 200         | 1000                     | 20%            |
| one hundred normal sized multi-blob transactions#02 | 200            | 100         | 100                      | 100%           |
| many small single share single blob transactions#03 | 300            | 200         | 500                      | 40%            |
| one hundred normal sized single blob transactions#03 | 200            | 100         | 100                      | 100%           |
| many single share multi-blob transactions#03      | 300              | 200         | 1000                     | 20%            |
| one hundred normal sized multi-blob transactions#03 | 200            | 100         | 100                      | 100%           |

with random enable those test result inconsistent for `many single share multi-blob transactions` and `one hundred normal sized multi-blob transactions` this is because the range between minsize and maxsize are higher than for the other tests case
Although the 2 other test cases result are the same with random 
#6012

**Comment by rawbytedev** [comment]
I added some improvement to test in #6077 and still researching for a way to solve this issue
Could I get some advice on it?
